src/backend/vulkan/src/lib.rs

```diff
@@ -687,17 +692,29 @@ impl adapter::PhysicalDevice<Backend> for PhysicalDevice {
                 0 => None,
                 1 => Some(*KHR_MAINTENANCE1),
                 _ => unreachable!(),
-            });
+            })
+            .chain(
+                if requested_extensions.contains(Features::TASK_SHADER) | requested_extensions.contains(Features::MESH_SHADER) {
+                    Some(*MESH_SHADER)
+                } else {
+                    None
+                }
+            );
 
         // Create device
         let device_raw = {
             let cstrings = enabled_extensions.map(CString::from).collect::<Vec<_>>();
 
             let str_pointers = cstrings.iter().map(|s| s.as_ptr()).collect::<Vec<_>>();
 
+            let mesh_shader_features = vk::PhysicalDeviceMeshShaderFeaturesNV::builder()
+                .task_shader(requested_extensions.contains(Features::TASK_SHADER))
+                .mesh_shader(requested_extensions.contains(Features::MESH_SHADER))
+                .build();
+
             let info = vk::DeviceCreateInfo {
                 s_type: vk::StructureType::DEVICE_CREATE_INFO,
-                p_next: ptr::null(),
+                p_next: &mesh_shader_features as *const vk::PhysicalDeviceMeshShaderFeaturesNV as *const std::ffi::c_void,
```

Comment:
just to clarify, will this confuse vulkan implementations that don't support or know about mesh shaders?

Reply:
Yeah, It might. I have reworked the code quite a lot now and I hope you like It. So now It uses the builder pattern from ash to be much cleaner and also assigns the p_next only if neccessary. I think this code structure will also make it much cleaner for any future extension.